art/estimators/classification/pytorch.py

```diff
@@ -701,3 +702,117 @@ def get_layers(self) -> List[str]:
 
         except ImportError:
             raise ImportError("Could not find PyTorch (`torch`) installation.")
+
+    def _apply_preprocessing_defences(self, x, y, fit: bool = False) -> Tuple[Any, Any]:
+        """
+        Apply all preprocessing defences of the estimator on the raw inputs `x` and `y`. This function is should
+        only be called from function `_apply_preprocessing`.
+
+        The method overrides art.estimators.estimator::BaseEstimator._apply_preprocessing_defences().
+        It requires all defenses to have a method `forward()`.
+        It converts numpy arrays to PyTorch tensors first, then chains a series of defenses by calling
+        defence.forward() which contains PyTorch operations. At the end, it converts PyTorch tensors
+        back to numpy arrays.
+
+        :param x: Samples.
+        :type x: Format as expected by the `model`
+        :param y: Target values.
+        :type y: Format as expected by the `model`
+        :param fit: `True` if the function is call before fit/training and `False` if the function is called before a
+                    predict operation.
+        :return: Tuple of `x` and `y` after applying the defences and standardisation.
+        :rtype: Format as expected by the `model`
+        """
+
+        if not hasattr(self, "preprocessing_defences") or \
+           self.preprocessing_defences is None or \
+           len(self.preprocessing_defences) == 0:
+            return x, y
+
+        if len(self.preprocessing_defences) == 1:
+            # Compatible with non-PyTorch defences if no chaining.
+            defence = self.preprocessing_defences[0]
+            x, y = defence(x, y)
+        else:
+            # Check if all defences are implemented in PyTorch.
+            for defence in self.preprocessing_defences:
+                if not hasattr(defence, "forward"):
```

Comment:
After introducing the new abstract base class `PreprocessorPyTorch` above we can use to test for the correct type here.